### PR TITLE
Add mipmap support, closes #21

### LINKED ISSUE #21: Mipmap Support
Just an oversight, we need to support mipmaps when uploading both cubemaps and regular textures.

### PR DESCRIPTION
**Pull request by Noxime**
Added a way to upload mip mapped textures. There is currently a problem in wgpu that requires a hack around the last mip level, but should be fixed soon.

### CODE REVIEW

File: examples/scene-viewer/src/main.rs [hunk 1]
```diff
@@ -74,12 +82,16 @@ fn load_skybox(renderer: &Renderer) -> Result<(), Box<dyn std::error::Error>> {
...
 
+    let mips = transcoder.get_total_image_levels(&file, 0);
+
     let mut prepared = transcoder
         .prepare_transcoding(&file)
         .ok_or("could not prepare skybox transcoding")?;
     let mut image = Vec::with_capacity(image_info.total_blocks as usize * 16 * 6);
-    for i in 0..6 {
-        image.extend_from_slice(&prepared.transcode_image_level(i, 0, basis::TargetTextureFormat::Bc7Rgba)?);
+    for mip in 0..mips {
```
Comment (cwfitzgerald) [inline examples/scene-viewer/src/main.rs]: Unfortunately, most tools lay out images the other way around `for layers { for mips { }}`, so in order to stay compatible with those tools (so you can just memcopy the data in), we need to remain consistent. This also means we can't copy all layers at once.

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
Thank you!

Everything looks fine except for the mip/layer ordering. Have you tested this on a mipmapped texture? I believe I made all my basis files have mipmaps, so you should be able to use them.

File: examples/scene-viewer/src/main.rs (hunk 1 above)
Comment (Noxime) [reply examples/scene-viewer/src/main.rs]: I tried this with the included skybox in scene-viewer, and it worked correctly

**Review by cwfitzgerald — APPROVED** [review]
Thank you!

**Pull request merged**